File: Langauge/Java/Reactive/02_Reactor/webfluxTuto/src/main/java/com/example/webfluxtuto/mapper/MemberMapper.java
package com.example.webfluxtuto.mapper;


import com.example.webfluxtuto.dto.MemberRequest;
import com.example.webfluxtuto.model.Member;
import org.apache.ibatis.annotations.Mapper;

import java.util.List;

@Mapper
public interface MemberMapper {

    List<Member> findAll();

    Member findById(int id);

    int insert(MemberRequest.Member member);

    int delete(long[] ids);

}

File: Langauge/Java/Framework/mybatis/basic/src/main/kotlin/com/example/mybatis/dao/MemberMapper.kt
package com.example.mybatis.dao

import com.example.mybatis.dto.MemberDTO
import org.apache.ibatis.annotations.Mapper

@Mapper
interface MemberMapper {

//    fun selectMember(): MemberDTO?

    fun selectAllMember(): List<MemberDTO>

    fun insertMember(member: MemberDTO): Int

    fun updateMember(member: MemberDTO): Int

    fun deleteMember(member: MemberDTO): Int

    fun selectByUserId(userId: String): MemberDTO?
}